test/currencyConvertor.ts

```diff
@@ -0,0 +1,73 @@
+import { ethers, waffle } from 'hardhat'
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import CurrencyConvertorArtifact from '../artifacts/contracts/proxies/CurrencyConvertor.sol/CurrencyConvertor.json';
+import ZeroExExchangeWrapperArtifact from '../artifacts/contracts/exchange-wrappers/ZeroExExchangeWrapper.sol/ZeroExExchangeWrapper.json';
+import StarkwareArtifact from '../artifacts/contracts/exchange-wrappers/Starkware.sol/Starkware.json';
+import UsdcContractArtifact from '../artifacts/contracts/tokens/usdcContract.sol/UsdcContract.json';
+
+const { deployContract } = waffle
+
+
+chai.use(solidity);
+const { expect } = chai;
+describe("CurrencyConvertor", () => {
+  let currencyConvertor: any;
+  let zeroExExchangeWrapper: any;
+  let starkware: any;
+  let usdc: any;
+  beforeEach(async () => {
+    const signers = await ethers.getSigners();
+
+    zeroExExchangeWrapper = await deployContract(
+      signers[0],
+      ZeroExExchangeWrapperArtifact,
+    );
+    starkware = await deployContract(
+      signers[0],
+      StarkwareArtifact,
+    );
+    usdc = await deployContract(
+      signers[0],
+      UsdcContractArtifact,
+    )
+
+    usdc.mint(
+      zeroExExchangeWrapper.address,
+      1000e6,
+    )
+    usdc.mint(
+      starkware.address,
+      1000e6,
+    )
```

Comment:
nit: These are transactions so they are promises and need to have `await`. Also, you can use `formatUnits(1000, 6)` instead of the constant `1000e6` (IMO that's more readable).

Reply:
hmmm seems to break the mint call